WordPress/Classes/ViewRelated/Reader/Subscriptions/ReaderSubscriptionsViewModel.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+import SwiftUI
+
+@MainActor
+final class ReaderSubscriptionsViewModel: ObservableObject {
+    private let store: CoreDataStackSwift
+
+    @Published private(set) var error: Error?
+    @Published private(set) var isRefreshing = false
+
+    private var refreshTask: Task<Void, Never>? {
+        didSet { isRefreshing = refreshTask != nil }
+    }
+
+    init(store: CoreDataStackSwift = ContextManager.shared) {
+        self.store = store
+    }
+
+    deinit {
+        refreshTask?.cancel()
+    }
+
+    func refresh() async {
+        if let task = refreshTask {
+            await task.value
+        }
+        let task = Task {
+            await _refresh()
+        }
+        refreshTask = task
+        return await task.value
+    }
+
+    private func _refresh() async {
+        error = nil
+        isRefreshing = true
+
+        await withUnsafeContinuation { continuation in
+            let service = ReaderTopicService(coreDataStack: self.store)
+            service.fetchAllFollowedSites(success: { [weak self] in
+                self?.refreshTask = nil
+                continuation.resume()
+            }, failure: { [weak self] error in
+                self?.error = error
+                self?.refreshTask = nil
+                continuation.resume()
+            })
+        }
```

Comment:
Nitpick: what do you think making the `withUnsafeContinuation` call return API result, and updating the `error` and `refreshTask` states in the context of the `async` function?

Reply:
Yeah, it's not ideal. I'll keep stuff async-first in the future PRs.